tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -999,6 +999,27 @@ def testMapCellsToColumns(self):
     self.assertEqual(columnsForCells[99], set([399]))
 
 
+  def testMaxSegmentsPerCellGetter(self):
+    tm = TemporalMemory(
+      columnDimensions=[64,64],
+      cellsPerColumn=32,
+      maxSegmentsPerCell=200
+    )
+    self.assertEqual(tm.connections.maxSegmentsPerCell, tm.getMaxSegmentsPerCell())
+    self.assertEqual(tm.connections.maxSegmentsPerCell, 200)
+    self.assertEqual(tm.getMaxSegmentsPerCell(), 200)
+
+  def testMaxSynapsesPerSegmentGetter(self):
+      tm = TemporalMemory(
+        columnDimensions=[32,32],
+        cellsPerColumn=16,
+        maxSynapsesPerSegment=150
+      )
+      self.assertEqual(tm.connections.maxSynapsesPerSegment, 150)
```

Comment:
I would remove this line. It tests the internal state of the object, which should be allowed to change. That is one of the purposes of the getter function itself.